specs/agents/tracing-sampling.md

```diff
@@ -1,12 +1,97 @@
+ - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0` means no transactions will be sampled
  - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0.5` means approximately 50% of transactions will be sampled
 
-If a transaction is not sampled, you should set the `sampled: false` property and omit collecting `spans` and `context`.
+##### Effect on metrics
+
+At the time of making a sampling decision,
+the sampling rate must be recorded so that it can be associated with every transaction and span in the trace.
+The sampling rate will be used by the server for scaling transaction and span metrics.
+
+Transaction metrics will be used by the UI to display transaction distributions and throughput,
+from the perspective of the transaction's service (grouped by `service.name` and `transaction.name`).
+
+Span metrics will be used by the UI for inter-service request rates on service maps,
+from the perspective of the requesting service (grouped by `service.name` and `destination.service.resource`).
+These are also referred as edge metrics.
+
+The server will calculate metrics by measuring only events from sampled traces,
+and scaling the metrics based on the sampling rate associated with each one.
+For example if the sampling rate is 0.5,
+then each sampled transaction and span would be counted twice in metrics aggregations.
+
+Metrics will be more accurate when the sampling rate is high.
+With lower sampling rates the server is able to calculate representative, but less accurate, metrics.
+If the sampling rate is 0 then no metrics will be calculated at all.
+
+Agents will record the sampling rate on transactions and spans as `sample_rate`. e.g.
+
+    {"transaction":{"name":"GET /","sample_rate":0.1,...}}
+    {"span":{"name":"SELECT FROM table","sample_rate":0.1,...}}
+
+For non-sampled transactions the `sample_rate` field _must_ be set to 0,
+to ensure non-sampled transactions are not counted in transaction metrics.
+This is important to avoid double-counting,
+as non-sampled transactions will be represented in metrics calculated from sampled transactions.
+
+When calculating transaction metrics,
+if the `sample_rate` transaction field is missing,
+the server will count each transaction (sampled and unsampled) as single events.
+This is required for backwards compatibility with agents that do not send a sampling rate.
+
+The server will only calculate span metrics for newer agents that include `sample_rate` in spans,
+as otherwise the representative counts will be incorrect for sampling rates less than 1.
+
+##### Non-sampled transactions
+
+Currently _all_ transactions are captured by Elastic APM agents.
+Sampling controls how much data is captured for transactions:
+sampled transactions have complete context recorded and include spans;
+non-sampled transactions have limited context and no spans.
+
+For non-sampled transactions set the transaction attributes `sampled: false` and `sample_rate: 0`, and omit `context`.
```

Comment:
How would we capture errors/outcomes for non-sampled transactions?

Reply:
Also we have outcomes on spans that can potentially impact the outcome of a transactions. Is that captured as a part of the limited context?

Reply:
The `outcome` is not nested under context but on the root level of the transaction, therefore it’s available even for non-sampled transaction